cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreEndpointContextInfo.java

```diff
@@ -0,0 +1,207 @@
+     * @param response the response to set the endpoint context for
+     */
+    public static void receivingResponse(OSCoreCtx oscoreCtx, Response response) {
+        setInfoIncoming(oscoreCtx, response);
+    }
+
+    /**
+     * Adds strings with information about the OSCORE context used to this
+     * destination endpoint context for outgoing messages.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param message the message to set information in the destination endpoint context for
+     */
+    private static void setInfoOutgoing(OSCoreCtx oscoreCtx, Message message) {
+
+        //Do nothing if this is not an instance of MapBasedEndpointContext (must cast to it)
+        if (message.getDestinationContext() instanceof MapBasedEndpointContext == false) {
+            return;
+        }
+
+        //Create new MapBasedEndpointContext for destination endpoint context with string values for OSCORE added
+        MapBasedEndpointContext endpointContext = (MapBasedEndpointContext) message.getDestinationContext();
+
+        MapBasedEndpointContext newEndpointContext = setInfo(oscoreCtx, endpointContext);
+
+        message.setDestinationContext(newEndpointContext);
+    }
+
+    /**
+     * Adds strings with information about the OSCORE context used to this
+     * source endpoint context for incoming messages.
+     *
+     * @param oscoreCtx the OSCORE context used for this message
+     * @param message the message to set information in the source endpoint context for
+     */
+    private static void setInfoIncoming(OSCoreCtx oscoreCtx, Message message) {
+
+        //Do nothing if this is not an instance of MapBasedEndpointContext (must cast to it)
+        if (message.getSourceContext() instanceof MapBasedEndpointContext == false) {
+            return;
+        }
+
+        //Create new MapBasedEndpointContext for source endpoint context with string values for OSCORE added
+        MapBasedEndpointContext endpointContext = (MapBasedEndpointContext) message.getSourceContext();
+
+        MapBasedEndpointContext newEndpointContext = setInfo(oscoreCtx, endpointContext);
+
+        message.setSourceContext(newEndpointContext);
+    }
+
+    /**
+     * Adds strings with information about the OSCORE context used to this
+     * endpoint context (creating a new one).
+     *
+     * @param oscoreCtx the OSCORE context used
+     * @param endpointContext the original endpoint context to set information based on
+     *
+     * @return the new endpoint context
+     */
+    private static MapBasedEndpointContext setInfo(OSCoreCtx oscoreCtx, EndpointContext endpointContext) {
+
+        //Create new MapBasedEndpointContext for this endpoint context with string values for OSCORE added
+        MapBasedEndpointContext newEndpointContext = MapBasedEndpointContext.addEntries(
+                endpointContext,
+                OSCORE, Boolean.TRUE.toString(),
```

Comment:
Consider to remove that "OSCORE, Boolean.TRUE.toString()" entry.
E.g. using `get(OSCORE_URI) != null` will do the same, or?

Reply:
True, that should provide the same functionality.

Reply:
Yes, as you wrote checking the existence of one of the other keys should provide the same functionality. I removed the OSCORE key in commit cbb3f2ed80dd3452f2b912d5cb1835cfc50120d4